program/src/matching.rs

```diff
@@ -704,6 +709,65 @@ impl<'a> Book<'a> {
         None
     }
 
+    /// Walk up the book and progresively find the best quantity and price to spend a given amount of quote.
+    pub fn get_best_order_for_amount(
+        &self,
+        side: Side,
+        quote_lot_amount_to_spend: i64,
+    ) -> Option<Order> {
+        let book_side = match side {
+            Side::Bid => self.bids.iter(),
+            Side::Ask => self.asks.iter(),
+        };
+        let mut cmlv_quantity: i64 = 0;
+        let mut best_price = 0; // Will update at each step, to settle on the the best price possible to spend the whole "quote_amount"
+        let mut quote_lot_left_to_spend = quote_lot_amount_to_spend;
+
+        for order in book_side {
+            // This order total value in quote lots
+            let order_size = order.quantity.checked_mul(order.price()).unwrap();
+            // How much base_lot we can fill for this order size
+            let quantity_matched = {
+                if quote_lot_left_to_spend < order_size {
+                    // we can finish the operation by purchasing this order partially
+                    // find out how much quantity that is in base lots
+                    quote_lot_left_to_spend.checked_div(order.price()).unwrap()
```

Comment:
you should decide here if you want to round down or up

Reply:
Updated

Reply:
Currently rounded down as we discussed, seems safer to have less